### PR TITLE
Ifpack2 Chebyshev: Add CG option for computation of spectral radius

### PR DESCRIPTION
**Pull request by cgcgcg**
@trilinos/ifpack2 

## Motivation
Add option to compute the spectral radius of D^-1 A using CG. Can be selected via "eigen-analysis: type" = "cg".

ML uses CG by default. This will simplify transitioning from ML to MueLu.

### CODE REVIEW

**Comment by cgcgcg** [comment]
@skennon10

**Review by jhux2 — COMMENTED** [review]
Using Belos is out of the question due to circular dependencies?

**Comment by cgcgcg** [comment]
I didn't want to try that. Also, we can save some memory and convergence checks by not using a Belos solver.

**Comment by jhux2** [comment]
Makes sense, would be good to have a unit test.

**Comment by cgcgcg** [comment]
That's why it's "DO NOT MERGE" :-)

File: packages/ifpack2/src/Ifpack2_Details_Chebyshev_decl.hpp [hunk 1]
```diff
@@ -64,6 +66,48 @@ template<class TpetraOperatorType>
 class ChebyshevKernel; // forward declaration
 #endif // DOXYGEN_SHOULD_SKIP_THIS
 
+template<class ScalarType, const bool lapackSupportsScalarType = LapackSupportsScalar<ScalarType>::value>
+struct LapackHelper{
+  static
+  ScalarType
+  tri_diag_spectral_radius(Teuchos::ArrayRCP<typename Teuchos::ScalarTraits<ScalarType>::magnitudeType> diag,
```
Comment (jhux2) [inline packages/ifpack2/src/Ifpack2_Details_Chebyshev_decl.hpp:73]: Can the LapackHelper struct be moved out of this `decl` file and into `Ifpack2_Details_Chebyshev_def.hpp`?

File: packages/ifpack2/src/Ifpack2_Details_Chebyshev_decl.hpp [hunk 2]
```diff
@@ -64,6 +66,48 @@ template<class TpetraOperatorType>
...
+    char char_N = 'N';
+
+    // lambdaMin = one;
+    ScalarType lambdaMax = STS::one();
+    if( N > 2 ) {
+      Teuchos::LAPACK<int,ScalarType> lapack;
+      lapack.PTEQR (char_N, N, diag.getRawPtr (), offdiag.getRawPtr (),
+                    &scalar_dummy, 1, &mag_dummy[0], &info);
+      TEUCHOS_TEST_FOR_EXCEPTION
+        (info < 0, std::logic_error, "Belos::PseudoBlockCGSolMgr::"
```
Comment (jhux2) [inline packages/ifpack2/src/Ifpack2_Details_Chebyshev_decl.hpp:100]: Update this message for Ifpack2.

**Review by jhux2 — CHANGES_REQUESTED** [review]

File: packages/ifpack2/src/Ifpack2_Details_Chebyshev_decl.hpp (hunk 1 above)
Comment (cgcgcg) [reply packages/ifpack2/src/Ifpack2_Details_Chebyshev_decl.hpp]: Done

File: packages/ifpack2/src/Ifpack2_Details_Chebyshev_decl.hpp (hunk 2 above)
Comment (cgcgcg) [reply packages/ifpack2/src/Ifpack2_Details_Chebyshev_decl.hpp]: Whoops :-) Done

**Review by jhux2 — APPROVED** [review]
lgtm

**Pull request merged**